### ISSUE TITLE
link-alt to fix card link accessibility not working for a page

### ISSUE THREAD

**Issue by bertiewooster** [executablebooks/sphinx-design#101]
### Describe the bug

**context**
When we make a page in RST with cards.

**expectation**
I expected each card link to have a discernible name.

**bug**
But instead there is no discernible name for the card link

**problem**
This is a problem for people using screen readers because the screen reader will not speak to them a meaningful name for the link.

This seems to be a bug with #78.

### Reproduce the bug

From the [RST source](https://github.com/sympy/sympy/pull/23746/files#diff-6f042ad8336579af372a07557f9ed88e475a2acb8a0521ebbdea40157d755b42), for the [page](https://output.circle-artifacts.com/output/job/262cb852-4776-403e-b631-d9510db39e10/artifacts/0/doc/_build/html/index.html) that is built, [Lighthouse still reports](https://googlechrome.github.io/lighthouse/viewer/?psiurl=https%3A%2F%2Foutput.circle-artifacts.com%2Foutput%2Fjob%2F262cb852-4776-403e-b631-d9510db39e10%2Fartifacts%2F0%2Fdoc%2F_build%2Fhtml%2Findex.html&strategy=desktop&category=performance&category=accessibility&category=best-practices&category=seo&category=pwa&utm_source=lh-chrome-ext) the issue  `Links do not have a discernible name`.

RST includes:

```
.. grid:: 2

   .. grid-item-card:: Tutorials :material-sharp:`school;3em;item-card-icon`
      :link: tutorials
      :link-type: ref
```

which seems to match the format given in the [sphinx-design documentation](https://sphinx-design.readthedocs.io/en/latest/grids.html#placing-a-card-in-a-grid).

**Comment by chrisjsewell** [comment]
Hey @bertiewooster, as mentioned in #78, this was closed in #89 by the addition of the `link-alt` option, have you tried using that?

**Comment by bertiewooster** [comment]
@chrisjsewell thanks. Oh, I didn't realize that a card would have to have a new option `link-alt`.

My thinking was that the card would automatically use the name of the card (in the example below, Tutorials) as the name for the link. That way, documentation writers wouldn't have to do anything additional to meet accessibility guidelines, thus making pages accessible by default.

As it's set up now, it seems like adding `link-alt` as in the following example would do the trick? (I didn't find `link-alt` in the [clickable cards documentation](https://sphinx-design.readthedocs.io/en/latest/cards.html?highlight=link-type#clickable-cards).)

```
   .. grid-item-card:: Tutorials :material-sharp:`school;3em;item-card-icon`
      :link: tutorials
      :link-type: ref
      :link-alt: Tutorials
```

cc @asmeurer

**Comment by chrisjsewell** [comment]
I believe this is fixed in #183
